CourseChamp/src/test/java/ca/mcgill/ecse428/CourseChamp/StepDefinitions/UserPortalStepDefinitions.java

```diff
@@ -0,0 +1,284 @@
+    public void the_admin_attempts_to_change_their_username(String string, String string2) {
+        // Write code here that turns the phrase above into concrete actions
+        ResponseEntity<AdminResponseDto> admin = (ResponseEntity<AdminResponseDto>)adminController.getAdminByEmail("/admin/"+string);
+        adminRequest = new AdminRequestDto();
+        adminRequest.setPassword(admin.getBody().getPassword());
+        adminRequest.setEmail(string);
+        adminRequest.setUsername(string2);
+
+        adminResponse = client.postForEntity("/admin/update", adminRequest, AdminResponseDto.class);
+    }
+
+    @When("the admin {string} unsuccessfully attempts to change their username to {string}")
+    public void the_admin_unsuccessfully_attempts_to_change_their_username(String string, String string2) {
+        // Write code here that turns the phrase above into concrete actions
+        ResponseEntity<AdminResponseDto> admin = (ResponseEntity<AdminResponseDto>)adminController.getAdminByEmail("/admin/"+string);
+        adminRequest = new AdminRequestDto();
+        adminRequest.setPassword(admin.getBody().getPassword());
+        adminRequest.setEmail(string);
+        adminRequest.setUsername(string2);
+
+        errorMessage = client.postForEntity("/admin/update", adminRequest, String.class);
+    }
+
+    @When("the admin {string} attempts to change their password to {string}")
+    public void the_admin_attempts_to_change_their_password(String string, String string2) {
+        // Write code here that turns the phrase above into concrete actions
+        ResponseEntity<AdminResponseDto> admin = (ResponseEntity<AdminResponseDto>)adminController.getAdminByEmail("/admin/"+string);
+        adminRequest = new AdminRequestDto();
+        adminRequest.setPassword(string2);
+        adminRequest.setEmail(string);
+        adminRequest.setUsername(admin.getBody().getName());
+
+        adminResponse = client.postForEntity("/admin/update", adminRequest, AdminResponseDto.class);
+    }
+
+    @When("the admin {string} unsuccessfully attempts to change their password to {string}")
+    public void the_admin_unsuccessfully_attempts_to_change_their_password(String string, String string2) {
+        // Write code here that turns the phrase above into concrete actions
+        ResponseEntity<AdminResponseDto> admin = (ResponseEntity<AdminResponseDto>)adminController.getAdminByEmail("/admin/"+string);
+        adminRequest = new AdminRequestDto();
+        adminRequest.setPassword(string2);
+        adminRequest.setEmail(string);
+        adminRequest.setUsername(admin.getBody().getName());
+
+        errorMessage = client.postForEntity("/admin/update", adminRequest, String.class);
+    }
+
+    @When("the student {string} attempts to view their reviews")
+    public void the_student_attemps_to_view_their_reviews(String string){
+        responseList = client.getForEntity("/student/getreviews/" + string, List.class);
```

Comment:
will need to add the endpoint "/student/getreviews/{email}" to ReviewController during our sprint